pkg/feature/templates/servicemesh/base/smcp.tmpl

```diff
@@ -0,0 +1,40 @@
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: {{ .Mesh.Name }}
+  namespace: {{ .Mesh.Namespace }}
+spec:
+  tracing:
+    type: None
+  addons:
+    grafana:
+      enabled: false
+    kiali:
+      name: kiali
+      enabled: false
+    prometheus:
+      enabled: false
+    jaeger:
+      name: jaeger
+  #security:
+  #  dataPlane:
+  #    mtls: true # otherwise inference-graph will not work. We use PeerAuthentication resources to force mTLS
```

Comment:
If these lines are included, inference would become blocked. The ingress displays the following in the access logs:

```
[2023-11-02T05:31:56.772Z] "POST /v2/models/sklearn-v2-iris/infer HTTP/2" 503 UF,URX upstream_reset_before_response_started{connection_failure,TLS_error:_336130315:SSL_routines:ssl3_get_record:wrong_version_number} - "TLS_error:_336130315:SSL_routines:ssl3_get_record:wrong_version_number" 178 212 60 - "10.128.3.215" "curl/8.0.1" "cf95a745-37e6-429f-8e28-2a11c86d1be3" "sklearn-v2-iris-predictor.kserve-test.svc.cluster.local" "10.128.3.215:8081" outbound|80||knative-local-gateway.istio-system.svc.cluster.local - 127.0.0.1:8443 127.0.0.1:35764 - -
```

I'm not sure what are the implications of commenting this out.

cc @RetoCode

Reply:
This should be included with `mtls: true`. If this blocks some connections, this means that either the caller or the callee is probably not part of the mesh (= does not have an service mesh sidecar). Our assumption with this is, that all communicating components are part of the mesh and use mTLS. This setting enforces that (which should be the goal).

Reply:
The issue seems to be fixed by using the TLS gateway you mentioned in the other  comment :-)